### PR TITLE
Limit Default Search to docker.io

### PR DESCRIPTION
**Pull request by adambkaplan**
Additional doc comments added from containers/image

### CODE REVIEW

File: imagecontent/registries.conf [hunk 1]
```diff
@@ -9,17 +9,10 @@
 # and 'registries.block'.
 
 [registries.search]
-registries = ['docker.io', 'registry.fedoraproject.org', 'quay.io', 'registry.access.redhat.com', 'registry.centos.org']
+# registries = ['docker.io', 'registry.fedoraproject.org', 'quay.io', 'registry.access.redhat.com', 'registry.centos.org']
```
Comment (bparees) [inline imagecontent/registries.conf]: i'm reasonably certain this is going to cause tests to fail unless we miraculously have fully qualified image specs for all our tests (I did recently go qualify a bunch, but for a long time we had stuff like "FROM centos").

but i guess we'll see.  Anyway you'll definitely want to investigate any aws-builds failures.

File: imagecontent/registries.conf [hunk 2]
```diff
@@ -9,17 +9,10 @@
...
 [registries.insecure]
 #registries = ['172.30.246.32:5000','172.30.1.1:5000','docker-registry.default.svc:5000','docker-registry.default.svc.cluster:5000','docker-registry.default.svc.cluster.local:5000']
 #registries = ['docker-registry.default.svc:5000','docker-registry.default.svc.cluster:5000','docker-registry.default.svc.cluster.local:5000','image-registry.openshift-image-registry.svc:5000','image-registry.openshift-image-registry.svc.cluster:5000','image-registry.openshift-image-registry.svc.cluster.local:5000']
-
-# If you need to block pull access from a registry, uncomment the section below
-# and add the registries fully-qualified name.
-#
-# Docker only
-[registries.block]
-registries = []
```
Comment (bparees) [inline imagecontent/registries.conf]: i'd rather leave this (commented out) to at least provide a hint about the syntax.

File: imagecontent/registries.conf [hunk 3]
```diff
@@ -9,17 +9,10 @@
...
 
 [registries.search]
-registries = ['docker.io', 'registry.fedoraproject.org', 'quay.io', 'registry.access.redhat.com', 'registry.centos.org']
+# registries = ['docker.io', 'registry.fedoraproject.org', 'quay.io', 'registry.access.redhat.com', 'registry.centos.org']
 
 # If you need to access insecure registries, add the registry's fully-qualified name.
 # An insecure registry is one that does not have a valid SSL certificate or only does HTTP.
 [registries.insecure]
 #registries = ['172.30.246.32:5000','172.30.1.1:5000','docker-registry.default.svc:5000','docker-registry.default.svc.cluster:5000','docker-registry.default.svc.cluster.local:5000']
 #registries = ['docker-registry.default.svc:5000','docker-registry.default.svc.cluster:5000','docker-registry.default.svc.cluster.local:5000','image-registry.openshift-image-registry.svc:5000','image-registry.openshift-image-registry.svc.cluster:5000','image-registry.openshift-image-registry.svc.cluster.local:5000']
```
Comment (bparees) [inline imagecontent/registries.conf]: you can update this to `registries = [ ]` too while you're at it.

**Comment by adambkaplan** [comment]
@bparees updated to default search to docker.io, per conversations in api-review with @smarterclayton.

**Comment by bparees** [comment]
/lgtm

keep in mind we'll still need to implement the defaulting behavior in the build controller too, so that if we overwrite this file (such as to set blocked registries) we still set the search registry to the default (if no other setting was provided)

**Pull request merged**